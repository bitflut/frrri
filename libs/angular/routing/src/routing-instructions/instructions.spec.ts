import { getActive } from './instructions/get-active.instruction';
import { getMany } from './instructions/get-many.instruction';

describe('instruction function', () => {
    it('should generate collection route', () => {
        const route = getMany();
        expect(route).toMatchSnapshot();
    });

    it('should generate active route', () => {
        const route = getActive();
        expect(route).toMatchSnapshot();
    });

    it('should generate collection route with parameters', () => {
        const route = getMany({
            params: {
                'filter[type][$eq]': 'comedy',
                'sort': 'startAt',
            },
        });
        expect(route).toMatchSnapshot();
    });
});
