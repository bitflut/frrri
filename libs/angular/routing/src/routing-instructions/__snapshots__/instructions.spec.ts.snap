// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`instruction function should generate active route 1`] = `
Object {
  "await": false,
  "awaitPlatformServer": true,
  "param": "id",
  "type": "getActive",
}
`;

exports[`instruction function should generate collection route 1`] = `
Object {
  "await": false,
  "awaitPlatformServer": true,
  "params": Object {},
  "type": "getMany",
}
`;

exports[`instruction function should generate collection route with parameters 1`] = `
Object {
  "await": false,
  "awaitPlatformServer": true,
  "params": Object {
    "filter[type][$eq]": "comedy",
    "sort": "startAt",
  },
  "type": "getMany",
}
`;
